diff --git a/smbus3/smbus3.py b/smbus3/smbus3.py
index a891755..dfab053 100644
--- a/smbus3/smbus3.py
+++ b/smbus3/smbus3.py
@@ -11,6 +11,7 @@ from ctypes import (
     c_uint8,
     c_uint16,
     c_uint32,
+    c_ulong,
     create_string_buffer,
     string_at,
 )
@@ -463,7 +464,7 @@ class SMBus:
 
         :rtype: int
         """
-        f = c_uint32()
+        f = c_ulong()
         ioctl(self.fd, I2C_FUNCS, f)
         return f.value
 
